code/crates/engine/src/sync.rs

```diff
@@ -283,8 +283,21 @@ where
             }
 
             Effect::GetDecidedValues(request_id, range, r) => {
+                // For simplicity, to avoid asking the application for the exact size of addresses, signatures, etc.
+                // we assume some maximum byte sizes for an address and a signature in order to calculate
+                // an **approximate** total size per value, so that we only send at most `self.sync_config.max_response_size`
+                // to another peer.
+                const MAX_BYTES_PER_ADDRESS: usize = 32;
+                const MAX_BYTES_PER_SIGNATURE: usize = 100;
```

Comment:
I would declare these variables close to the declarations of the address and signature types. Also, each app will have its own definition of address and signature.

Reply:
Not sure I fully follow your comment: Are you suggesting contacting the app to receive the sizes of addresses and signatures? If not, where exactly would you declare those constants?

Reply:
We could add them as static const methods to the `Address` and `SigningScheme` traits.

Reply:
I am on it.

Reply:
But that's pretty brittle as well, and could break things if the upper bound is not tight enough, eg. some application uses a signature scheme with variable-size signatures and defines `Signature::MAX_SIZE` to a large value just in case.

Reply:
The best way would probably be to encode the commit certificate (or event the full sync response) using the `(Network)Codec` and use the actual computed size. Not great for performance, we should probably either measure the impact upfront, or go with it and only optimize this if we see too big a perf impact. What do you think?